doc/Witch/intro-to-witch.md

```diff
@@ -0,0 +1,127 @@
+specification for such an assistant for Juvix uses algebraic effects and
+handlers as the means of congregation.
+
+Our approach is named Witch, a play on the
+assistant tools colloquially called "wizards". There is no consensus of what a
+wizard is or what exactly the tasks are it is supposed assist with. Wizards seem
+to be used mostly for multiple-step and/or configuration features, however. We
+went for the name "witch" to align it to the idea of assistant tools,
+while dodging the overloaded and confusing terminology.
+
+We propose an effects and handlers view of such proofs,
+based on prior work developed on the [Andromeda
+prover](https://www.andromeda-prover.org).  
+In Juvix, you program as you would normally and
+invoke the proof environment as an effect to prove certain properties as
+you go. Given that the proof environment is *just* an effect, different proof
+styles (e.g., Agda-style dependent types , 
+SMT solver, proof search) can be _composed_ under a shared interface, a
+proof object that can manipulate itself while querying different
+automated provers.
+
+The reasons we employ algebraic effects and handlers are numerous:
+
+1.  as proofs cannot be fully automated, all approaches that try to
+    automate the process (e.g, proof search, SMT solver) may be
+    non-deterministic or never find a solution. Therefore, the system
+    should be able to handle "impure" computations and errors. Algebraic
+    effects and handlers have well-defined semantics and provide a
+    simple interface for performing effects. With them, we avoid
+    indiscriminate effects that are often error-prone and intricate
+    effects machinery such as monad transformers;
+
+2.  the semantics of effects and handlers accommodate composition of
+    arbitrary effects and the composition of multiple handlers, which
+    means users have the ability to weaken more general strategies into
+    specific ones while maintaining top-level handlers unmodified;
+
+3.  effects and handlers have well-defined semantics, and it is the main
+    feature of many new languages, such as
+    [Koka](https://koka-lang.github.io/koka/doc/index.html),
+    [Eff](https://www.eff-lang.org), [Frank](https://github.com/frank-lang) and 
+    [Effekt](https://effekt-lang.org), and have implementation in all major
+    functional programming languages, what guarantees that our approach is based
+    in research and real-world experiences, and not something we hacked together
+    over the weekend. 
+
+## The Essence of Witch
+
+As for the syntax, we use `operation { params } via handler`, which is
+a syntactic improvement over `handler(operation, params)`, since effect handling
+is similar to function application, but also carries effect information.
+The user defines data types and functions as usual, and then uses
+Witch to prove properties about said definitions. The Examples below
+shows it in action.
+
+![](paper/image/witch.svg)
```

Comment:
This didn't show in the rendered version?

Reply:
Err, forgot to add when moved it away from witch's repo. Thanks for catching it.